### DISCUSSION TITLE
Duplicate messages in a stream queue after reconnection with AutomaticRecoveryEnabled

### DISCUSSION THREAD

**Discussion by Laurianti** [rabbitmq/rabbitmq-dotnet-client#1315] [General]
I have encountered an issue with the RabbitMQ .NET client when working with stream queues and using the auto-recovery feature (AutomaticRecoveryEnabled = true). After a reconnection, the client resumes reading messages from the starting offset specified by the consumer, rather than the offset it was at before the disconnection, causing duplicate messages.

Steps to reproduce:

1. Create a stream queue in RabbitMQ Server.
2. Register a consumer for the queue using the RabbitMQ .NET client and set the starting offset 0.
3. Publish some messages into the queue and wait for the consumption.
4. Restart the RabbitMQ node.
5. Wait for the client to reconnect automatically.
6. Note that the client resumes reading messages from the offset 0 again, rather than the offset it was at before the disconnection, causing duplicate messages.

**Expected behavior**: after the client reconnects, the RabbitMQ .NET client should resume reading messages from the offset it was at before the disconnection, and should not consume messages that were previously consumed.

**Actual behavior**: after the client reconnects, the RabbitMQ .NET client resumes reading messages from the starting offset specified by the consumer (offset 0), rather than the offset it was at before the disconnection, causing duplicate messages.

I expect the messages to be acknowledged and not consumed again.

Versions used:

RabbitMQ .NET client: 6.4.0
RabbitMQ Server: 3.10.2

Thank you in advance for your support in resolving this issue.

**Comment by michaelklishin** [comment]
This client is an AMQP 0-9-1 one. It can consume from streams but with plenty of limitations.

If you want access to every stream feature and stream-oriented design decisions, see this [RabbitMQ Streams client for .NET](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client).

**Comment by lukebakken** [comment]
@Laurianti your use-case is why this feature was added: https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1304. This feature will ship in version 6.5.0 of the library.

You must be using version 6.4.0 because that is the latest version.

cc @Zerpet

**Comment by Laurianti** [comment]
Sorry, you are right, the version is the 6.4.0
Thank you for your answer, I will try
